### PR TITLE
Add remaining bindings for the Intl namespace

### PR DESCRIPTION
**Pull request by thomaseizinger**
This PR adds remaining bindings for the Intl namespace:

- NumberFormat
- PluralRules

### CODE REVIEW

File: crates/js-sys/src/lib.rs [hunk 1]
```diff
@@ -3433,13 +3433,105 @@ pub mod Intl {
...
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/format
+        #[wasm_bindgen(method, getter, js_class = "Intl.NumberFormat")]
+        pub fn format(this: &NumberFormat) -> Function;
+
+        /// The Intl.Numberformat.prototype.formatToParts() method allows locale-aware
+        /// formatting of strings produced by NumberTimeFormat formatters.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/formatToParts
+        #[wasm_bindgen(method, js_class = "Intl.NumberFormat", js_name = formatToParts)]
+        pub fn format_to_parts(this: &NumberFormat, number: f64) -> Array;
```
Comment (thomaseizinger) [inline crates/js-sys/src/lib.rs]: Not sure if the `f64` here is correct.

File: crates/js-sys/src/lib.rs [hunk 2]
```diff
@@ -3433,13 +3433,105 @@ pub mod Intl {
...
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/PluralRules/resolvedOptions
+        #[wasm_bindgen(method, js_namespace = Intl, js_name = resolvedOptions)]
+        pub fn resolved_options(this: &PluralRules) -> Object;
+
+        /// The Intl.PluralRules.prototype.select method returns a String indicating
+        /// which plural rule to use for locale-aware formatting.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/PluralRules/select
+        #[wasm_bindgen(method, js_namespace = Intl)]
+        pub fn select(this: &PluralRules, number: f64) -> JsString;
```
Comment (thomaseizinger) [inline crates/js-sys/src/lib.rs]: Same here. I tried it in the browser and it accepts floating point numbers.

File: crates/js-sys/src/lib.rs (hunk 1 above)
Comment (fitzgen) [reply crates/js-sys/src/lib.rs]: Looks correct to me!

**Comment by thomaseizinger** [comment]
@fitzgen I guess you can now check the boxes in the issue to update the progress of implementing the bindings!

**Comment by fitzgen** [comment]
> @fitzgen I guess you can now check the boxes in the issue to update the progress of implementing the bindings!

Done! Thanks for the reminder :)

**Pull request merged**